src/main/java/org/reaktivity/nukleus/mqtt/internal/MqttUtility.java

```diff
@@ -0,0 +1,33 @@
+/**
+ * Copyright 2016-2019 The Reaktivity Project
+ *
+ * The Reaktivity Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package org.reaktivity.nukleus.mqtt.internal;
+
+import java.util.regex.Pattern;
+
+public final class MqttUtility
+{
+    public static final Pattern TOPIC_FILTER_REGEX = Pattern.compile("[/]?(([^/#+]*|\\+)/)*(#|\\+|[^/#+]*)");
+
+    public static boolean validTopicFilter(String topicFilter)
+    {
+        return TOPIC_FILTER_REGEX.matcher(topicFilter).matches();
+    }
+
+    private MqttUtility()
+    {
+        // Utility
+    }
+}
```

Comment:
Alternatively, rename this class to `MqttValidator`, make the `isTopicFilterValid(topicFilter)` method non-static, eliminate the need for `ThreadLocal`, but use the same approach to reuse the `Matcher` and create an instance of `MqttValidator` in `MqttServerFactory` that is reused by all `mqtt` streams on the same thread (a different thread will have its own `MqttServerFactory`).